osidb/models.py

```diff
@@ -1753,6 +1767,25 @@ def create_flawcomment(flaw, external_system_id, comment, **extra_fields):
             flawcomment.meta_attr = comment
             return flawcomment
         except ObjectDoesNotExist:
+            # The "text" field is required for full validation, so save it if the caller forgets
+            # to explicitly copy it over from meta_attr.
```

Comment:
How is it meant that the caller could forget to copy from  `meta_attr`? I would expect that the `text` field to be vital here and the `meta_attr` is rather a some extra/debugging/optional information.

Reply:
I agree `text` is vital, hence the lines 1770-1775. This became necessary with the addition of `BugzillaSyncMixin` to `FlawComment`'s ancestors, since `BugzillaSyncMixin` forces `validate()` on `save()`.

AFAIK `meta_attr` is actually the primary and original set of information from bugzilla. Parts of it are duplicated into normal model fields. Here's the breakdown:

* the dict that is stored into `meta_attr` https://github.com/RedHatProductSecurity/osidb/blob/f41cf220cbdf0cb197dc404b9fef48996e5d0393/osidb/models.py#L1759 comes from the 3rd argument `comment`
* this `comment` dictionary comes from a list in `[FlawConvertor instance].flaw_comments` https://github.com/RedHatProductSecurity/osidb/blob/f41cf220cbdf0cb197dc404b9fef48996e5d0393/collectors/bzimport/convertors.py#L705 which comes from `self._flaw_comments` which is set in `__init__()`
* that is provided in https://github.com/RedHatProductSecurity/osidb/blob/f41cf220cbdf0cb197dc404b9fef48996e5d0393/collectors/bzimport/collectors.py#LL460C17-L460C17
* and that is loaded from bugzilla in https://github.com/RedHatProductSecurity/osidb/blob/f41cf220cbdf0cb197dc404b9fef48996e5d0393/collectors/bzimport/collectors.py#L97

If you try to delete line 1772 and rerun tests, you will see an example; I think it's `test_flaw.py:121`.

Reply:
I am not sure I get it but it is not such a big deal. You can leave it as it is.

Reply:
Perhaps by `caller` you mean an API user, whereas I meant it as a caller (i.e. python code) of this method. So I'll change the comment to say "caller of this method" to be clear.

Reply:
Resolved in https://github.com/RedHatProductSecurity/osidb/pull/209/commits/f9ebfa5badd141532da083c3d6506b796e4868b4